tests/console/snapper_cleanup_timeline.pm

```diff
@@ -0,0 +1,61 @@
+# SUSE's openQA tests
+#
+# Copyright © 2020 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Configure snapper and verify that timeline cleanup algorithm behaves accordingly.
+# Maintainer: QA SLE yast team <qa-sle-yast@suse.de>
+
+use strict;
+use warnings;
+use base 'y2_installbase';
+use testapi;
+use scheduler 'get_test_suite_data';
+use Test::Assert ':all';
+
+sub pre_run_hook {
+    my $test_data = get_test_suite_data();
+    select_console 'root-console';
+    record_info("Configuration", "Configure snapper.");
+    foreach my $param (keys %{$test_data->{snapper_config}}) {
+        assert_script_run("snapper set-config $param=$test_data->{snapper_config}->{$param}",
+            fail_message => "Snapper configuration failed for parameter $param");
+    }
+}
+
+sub run {
+    record_info("Check quota", "Verify that the percentage of root filesystem quota is less than 50%");
+    my $convert_numeric = qr/[^0-9.]+/;
+    my $disk_space      = script_output("df -h /  | awk \'NR==2 {print \$4}\'");
+    $disk_space =~ s/$convert_numeric//;
+    my $qgroup_space = script_output("btrfs qgroup show -F --sync --si / | awk \'NR==3 {print \$2}\'");
+    $qgroup_space =~ s/$convert_numeric//;
+    die "Root filesystem quota take more than 50% of root disk space" if ($qgroup_space >= 0.5 * $disk_space);
+
+    record_info("Space > 20%", "Ensure that free space is more than 20%. 
+	    Create a snapshot with timeline cleanup algorithm and make sure that it is not erased by the algorithm");
+    $disk_space = script_output("df -h /  | awk \'NR==2 {print \$5}\'");
```

Comment:
same apply here to strip the unit part, could you create a function to not repeat this line in the test 4 times?

Reply:
Done